ast/compile.go

```diff
@@ -3978,9 +4010,43 @@ func rewriteDeclaredVarsInBody(g *localVarGenerator, stack *localDeclaredVars, u
 		cpy.Append(NewExpr(BooleanTerm(true)))
 	}
 
+	errs = checkUnusedAssignedVars(body[0].Loc(), stack, used, errs, strict)
 	return cpy, checkUnusedDeclaredVars(body[0].Loc(), stack, used, cpy, errs)
 }
 
+func checkUnusedAssignedVars(loc *Location, stack *localDeclaredVars, used VarSet, errs Errors, strict bool) Errors {
+
+	if !strict || len(errs) > 0 {
+		return errs
+	}
+
+	dvs := stack.Peek()
+	unused := NewVarSet()
+
+	for v, occ := range dvs.occurrence {
+		if !v.IsWildcard() && occ == assignedVar && stack.Count(v) <= 1 {
```

Comment:
Do we need to treat "count == 1" as unused..? 🤔 that will perhaps some day in the future surprise me a little.

Reply:
I can add a comment here. Since the var was assigned in this scope, it also needs to have been seen more than once (time of assignment) in this, and nested, scopes.

Reply:
Ah, assignment is considered use, of course. That makes sense, I was too caught up thinking about _declarations_ instead of _assignments_. 👍